### PR TITLE
LPS-113529 Modernize the `getLexiconIcon()` AUI utility

### PR DESCRIPTION
**Pull request by kresimir-coko**
LPS: https://issues.liferay.com/browse/LPS-113529

The utility itself wasn't old in the sense that it had any AUI code, but I still went with moving it to `frontend-js-web` so that we can add types to the TS definition file, tests, and improve the code slightly.

The biggest change was removing the reliance on `Liferay.Util.getLexiconIconTpl()` and `Liferay.Util.sub` (by extension) by adding a new function inside the new util that handles what those 2 utils did, in this case.

### CODE REVIEW

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/get_lexicon_icon.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+				aria-hidden="true"
+				class="lexicon-icon lexicon-icon-${iconName} ${cssClass}"
+				focusable="false"
+				role="presentation"
+			>
+				<use href="${themeDisplay.getPathThemeImages()}/clay/icons.svg#${iconName}" />
+			</svg>`;
+	}
+
+	const lexiconIconTemplate = getLexiconIconTemplate(icon, cssClass);
```
[the remark this answers is not in the document]
Comment (kresimir-coko) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/get_lexicon_icon.js:31]: The next task in this epic is to handle the [Util.getLexiconIconTpl](https://github.com/liferay-frontend/liferay-portal/blob/748834878182fee50e1b258af8ae0779880c882f/modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util.js#L458) where I was going to tackle that exact problem. 

I think that it's best to replace them bit by bit even if I am going to have to change the `getLexiconIcon` util again to use the new `getLexiconIconTpl` utility, just to keep PRs small and tidy.

**Comment by kresimir-coko** [comment]
ci:test:relevant

**Comment by kresimir-coko** [comment]
ci:test:relevant

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/get_lexicon_icon.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	const tempElement = document.createElement('div');
+
+	tempElement.innerHTML = lexiconIconTemplate;
+
+	return tempElement.firstChild;
```
Comment (markocikos) [inline modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/get_lexicon_icon.js:36]: Should we use a fragment here? Just a thought, I don't think it makes much difference in this case.

**Review by markocikos — COMMENTED** [review]
Few minor things inside, mostly looks good!

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/get_lexicon_icon.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (diegonvs) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/get_lexicon_icon.js:31]: > Just curious, do we have a modernized equivalent of A.Lang.sub, that would work with template literals?

Good question, I think not. What I've been seeing around DXP when people try to use template literals is interpolating strings there and then calling a `.join` to remove whitespaces. 

Does make sense bringing a more useful utility and avoid concatenating html?

[the remark this answers is not in the document]
Comment (kresimir-coko) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/get_lexicon_icon.js:31]: @markocikos @diegonvs 
> function `getLexiconIconTemplate` does not actually get the template...

True! Gonna rename it

> Do we have a modernized equivalent of `A.Lang.sub``

We have [Liferay.Util.sub](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util.js#L923), that I will move over to `frontend-js-web` as part of this epic

**Comment by kresimir-coko** [comment]
ci:test:relevant

**Comment by kresimir-coko** [comment]
@markocikos I've renamed the function, so if you could please just give it a quick glance before I forward 🙏

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/get_lexicon_icon.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (markocikos) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/get_lexicon_icon.js]: > Do we have a modernized equivalent of `A.Lang.sub``
> 
> We have [Liferay.Util.sub](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util.js#L923), that I will move over to frontend-js-web as part of this epic

I believe these two are different utilities, it's unfortunate that they are named very similarly:
- The purpose of `Liferay.Util.sub` is to replace placeholder in language keys. The `x` in something like `from-x-to-x`. This is Liferay-specific.
- `A.Lang.sub` is YUI-native templating mechanism. It requires HTML template string with placeholders in brackets, like `<source srcset="{srcset}" media="{media}">`

The first one is easy to modernize. For the second one, I don't think we have an equivalent.

**Review by markocikos — APPROVED** [review]
LGTM!

**Comment by kresimir-coko** [comment]
ci:forward

**Pull request closed without merging**